File: examples/reliability/example_reliability.R
library("psych")
bfi2 <- mmstat4::ghload("data/bfi4.rds")
# extract first factor
fa   <- fa(bfi2)
vars <- abs(fa$loadings)>0.5
# Cronbachs alpha (items not reversed)
alpha(cor(bfi2[,vars]))
# Cronbachs alpha (items reversed)
alpha(cor(bfi2[,vars]), check.keys=T)
# Cronbachs alpha and sum scores
keys <- ifelse (vars, sign(fa$loadings)<0, 0)
si   <- scoreItems(keys, bfi2)

File: examples/reliability/example_reliability.py
import pandas as pd
import pingouin as pg
bfi2 = r['mmstat4::ghload']('bfi4.rds')
df   = pd.DataFrame(bfi2.iloc[:, :5])
# Compute Cronbach's Alpha using pingouin
alpha, alpha_ci = pg.cronbach_alpha(df)
print(f"Cronbach's Alpha: {alpha:.3f}")
# compute sum scores
df['sum_score'] = df.sum(axis=1)
df.head()
